x/fbridge/keeper/keeper.go

```diff
@@ -55,6 +59,104 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", "x/"+types.ModuleName)
 }
 
+func (k Keeper) InitMemStore(ctx sdk.Context) {
+	memStore := ctx.KVStore(k.memKey)
+	memStoreType := memStore.GetStoreType()
+	if memStoreType != sdk.StoreTypeMemory {
+		panic(fmt.Sprintf("invalid memory store type; got %s, expected: %s", memStoreType, sdk.StoreTypeMemory))
+	}
+
+	// create context with no block gas meter to ensure we do not consume gas during local initialization logic.
+	noGasCtx := ctx.WithBlockGasMeter(sdk.NewInfiniteGasMeter())
+
+	roleMetadata := types.RoleMetadata{}
+	if !k.IsInitialized(noGasCtx) {
+		for _, pair := range k.GetRolePairs(noGasCtx) {
+			switch pair.Role {
+			case types.RoleGuardian:
+				roleMetadata.Guardian++
+			case types.RoleOperator:
+				roleMetadata.Operator++
+			case types.RoleJudge:
+				roleMetadata.Judge++
+			}
+		}
+		k.setRoleMetadata(noGasCtx, roleMetadata)
+
+		nInactive := uint64(0)
+		for _, bs := range k.GetBridgeSwitches(ctx) {
+			if bs.Status == types.StatusInactive {
+				nInactive++
+			}
+		}
+		k.setBridgeInactiveCounter(noGasCtx, nInactive)
+
+		memStore := noGasCtx.KVStore(k.memKey)
+		memStore.Set(types.KeyMemInitialized, []byte{1})
+	}
+}
+
+// IsInitialized returns true if the keeper is properly initialized, and false otherwise.
+func (k Keeper) IsInitialized(ctx sdk.Context) bool {
+	memStore := ctx.KVStore(k.memKey)
+	return memStore.Get(types.KeyMemInitialized) != nil
+}
+
 func (k Keeper) GetAuthority() string {
 	return k.authority
 }
+
+func (k Keeper) IsBridgeHalted(ctx sdk.Context) bool {
+	return k.GetBridgeStatus(ctx) == types.StatusInactive
+}
+
+func (k Keeper) setRoleMetadata(ctx sdk.Context, data types.RoleMetadata) {
+	memStore := ctx.KVStore(k.memKey)
+	bz := k.cdc.MustMarshal(&data)
+	memStore.Set(types.KeyMemRoleMetadata, bz)
+}
+
+func (k Keeper) GetRoleMetadata(ctx sdk.Context) types.RoleMetadata {
+	memStore := ctx.KVStore(k.memKey)
+	data := types.RoleMetadata{}
+	bz := memStore.Get(types.KeyMemRoleMetadata)
+	if bz == nil {
+		return types.RoleMetadata{}
+	}
+	k.cdc.MustUnmarshal(bz, &data)
+	return data
+}
+
+func (k Keeper) GetBridgeStatus(ctx sdk.Context) types.BridgeStatus {
+	roleMeta := k.GetRoleMetadata(ctx)
+	bsMeta := k.GetBridgeStatusMetadata(ctx)
+	if types.CheckTrustLevelThreshold(roleMeta.Guardian, bsMeta.Inactive, k.GetParams(ctx).GuardianTrustLevel) {
+		return types.StatusInactive
+	}
+
+	return types.StatusActive
+}
```

Comment:
Is it okay for `GetBridgeStatus()` to return `types.StatusActive` when `roleMeta.Guardian is 0` and `bsMeta.Inactive is 0`?

Reply:
thank you for finding edge case!